onnxruntime/test/providers/tensorrt/tensorrt_basic_test.cc

```diff
@@ -87,6 +88,190 @@ void CreateBaseModel(std::string model_name, std::string graph_name, std::vector
   status = onnxruntime::Model::Save(model, model_name);
 }
 
+void RunSession(InferenceSession& session_object,
```

Comment:
we might consider in the future to migrate these to inference_session_test ?
these tests are basically generic and could be applied to all providers.
probably don't do it in this PR before testing whether all EP's can pass these tests.